client/src/app/models/input-property.model.ts

```diff
@@ -0,0 +1,16 @@
+/** Change the type when we have the models imported */
+export class InputProperty {
+    public id: string;
+    public type: string;
+    public value: string;
+
+    constructor(attr: {
+        id?: string;
+        type?: string;
+        value?: string;
+    }) {
+        this.id = attr.id || "";
+        this.type = attr.type || "";
+        this.value = attr.value || "";
+    }
+}
```

Comment:
@mayacoda  
Is
```javascript
{
  id: "",
  type: "",
  value: ""
}
``` 
a valid InputProperty?
Would this be better off as an immutable value object identified by its `id`?